File: carrinho/carroShow/application.windows32/source/Carro.py
import Constantes as CTE
class Carro:
    largura = 0.1
    comprimento = 0.1
    img = None
    def __init__(self,c,p):
        self.c = c
        self.p = p
        
        self.alive = True
        
    def show(self):
        if not self.alive :
            return
        angle = atan2(self.p.y - self.c.y,self.p.x - self.c.x) 
        with push():
            translate(self.c.x*CTE.SCL,self.c.y*CTE.SCL)
            rotate(angle);
            rect(-Carro.largura/2*CTE.SCL,-Carro.comprimento/2*CTE.SCL,Carro.largura*CTE.SCL,Carro.comprimento*CTE.SCL)
            circle(0,-Carro.comprimento/2*CTE.SCL,0.02*CTE.SCL)    
            #image(Carro.img,-Carro.largura/2*CTE.SCL,-Carro.comprimento/2*CTE.SCL,Carro.largura*CTE.SCL,Carro.comprimento*CTE.SCL)
            

File: projeto sumocup/carrinho/showWeb/Carro.js
var im = 0;
class Carro {
  constructor(c, p) {      
    this.c = c;
    this.p = p;
    this.alive = true;
    this.img = imgs[im];
    im++;
    if(im==2)im=0;
  }
  toString() {
    return "c(%.2f,%.2f) p(%.2f,%.2f)%s".format(this.c.x, this.c.y, this.p.x, this.p.y, this.alive ? "vivo" : "morto");
  }
  update(c,p,alive){
    this.c = c;
    this.p = p;
    this.alive = alive;
  }
  show() {
    //if (!this.alive) { return null;}
    //console.log(this.c);
   let angle = atan2(-this.p.y + this.c.y,this.p.x - this.c.x)+ PI/2;
    push();
    if(this.alive){stroke(color(255,0,0));}
    translate(this.c.x*SCL, -this.c.y*SCL);
    rotate(angle);
    //rect(-largura/2*SCL, -comprimento/2*SCL, largura*SCL, comprimento*SCL);
    image(this.img,-largura/2*SCL, -comprimento/2*SCL, largura*SCL, comprimento*SCL);
    circle(0, -comprimento/2*SCL, 0.02*SCL);
    pop();

  }
}
